Source/NavierStokesBase.cpp

```diff
@@ -3234,58 +3253,30 @@ NavierStokesBase::SyncInterp (MultiFab&      CrseSync,
     //
     //////////
     // Should be fine for EB for now, since EB doesn't intersect Phys BC
-    // Although calling filcc_tile is not what we really want to be doing,
-    // there must be something higher level
-    // Maybe setPhysBoundaryValues? see example below
-    //
-    // MultiFab test(cdataBA,fdmap,num_comp,0,MFInfo(),Factory());
-    // // replace divu with test and compare with cdataMF?
-    // divu_old.FillBoundary();
-    // Real prev_time = amr_level.get_state_data(Divu_Type).prevTime();
-    // for (MFIter mfi(divu_new); mfi.isValid(); ++mfi)
-    // {
-    //   amr_level.setPhysBoundaryValues(divu_old[mfi],Divu_Type,prev_time,0,0,1);
-    // }
+    // Not sure about what happens if EB intersects Phys BC
     ///////
 
     // tiling may not be needed here, but what the hey
-
+    GpuBndryFuncFab<DummyFill> gpu_bndry_func(DummyFill{});
 #ifdef _OPENMP
-#pragma omp parallel
+#pragma omp parallel if (Gpu::notInLaunchRegion())
 #endif
+    for (MFIter mfi(cdataMF,TilingIfNotGPU()); mfi.isValid(); ++mfi)
     {
-      int* bc_new = new int[2*AMREX_SPACEDIM*(src_comp+num_comp)];
-      for (MFIter mfi(cdataMF,true); mfi.isValid(); ++mfi)
-      {
+       const Box& bx   = mfi.tilebox();
+       FArrayBox& data = cdataMF[mfi];
 
-        FArrayBox&  cdata   = cdataMF[mfi];
-        const int*  clo     = cdata.loVect();
-        const int*  chi     = cdata.hiVect();
-        const Box&  bx      = mfi.tilebox();
-        RealBox     gridloc = RealBox(bx,fine_level.geom.CellSize(),fine_level.geom.ProbLo());
-        const int*  lo      = bx.loVect();
-        const int*  hi      = bx.hiVect();
-        const Real* xlo     = gridloc.lo();
-
-        for (int n = 0; n < num_comp; n++)
-        {
-          set_bc_new(bc_new,n,src_comp,lo,hi,cdomlo,cdomhi,cgrids,bc_orig_qty);
-          filcc_tile(ARLIM(lo),ARLIM(hi),
-                     cdata.dataPtr(n), ARLIM(clo), ARLIM(chi),
-                     cdomlo, cdomhi, dx_crse, xlo,
-                     &(bc_new[2*AMREX_SPACEDIM*(n+src_comp)]));
-        }
-      }
-      delete [] bc_new;
+       Vector<BCRec> bx_bcrec(num_comp);
+       set_bcrec_new(bx_bcrec,num_comp,src_comp,bx,cdomain,cgrids,bc_orig_qty);
+       gpu_bndry_func(bx,data,0,num_comp,geom,0.0,bx_bcrec,0,0);
```

Comment:
Need `geom` -> `cgeom` here since cdataMF lives on the coarse level.

Reply:
More specifically, so that it gets the right domain box, dx into the functions below (even though here it may not actually matter...).